### PR TITLE
[Capture] JIT entire workflows with default qubit

### PR DESCRIPTION
**Pull request by albi3ro**
**Context:**

When messing around with #6596, I realized we weren't able to jit the full workflow even though the device execution itself can still be jitted. I needed to register the mlir lowering to just be the lowering of the implementation function. 

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-79566]

### CODE REVIEW

File: tests/capture/test_capture_mid_measure.py [hunk 1]
```diff
@@ -309,7 +309,7 @@ class TestMidMeasureExecute:
 
     @pytest.mark.xfail(strict=False)  # single branch statistics sometimes gives good results
     @pytest.mark.parametrize("reset", [True, False])
-    @pytest.mark.parametrize("postselect", [None, 0, 1])
+    @pytest.mark.parametrize("postselect", [pytest.param(None, marks=pytest.mark.xfail), 0, 1])
```
Comment (albi3ro) [inline tests/capture/test_capture_mid_measure.py:312]:
```suggestion
    @pytest.mark.parametrize("postselect", [None, 0, 1])
```

File: pennylane/workflow/_capture_qnode.py [hunk 2]
```diff
@@ -298,6 +300,8 @@ def _qnode_jvp(args, tangents, **impl_kwargs):
 
     batching.primitive_batchers[qnode_prim] = _qnode_batching_rule
 
+    mlir.register_lowering(qnode_prim, mlir.lower_fun(qnode_impl, multiple_results=True))
```
Comment (mudit2812) [inline pennylane/workflow/_capture_qnode.py:303]: What is the default lowering rule, and why did that not work in our case?

Comment (albi3ro) [reply pennylane/workflow/_capture_qnode.py:303]: The default lowering rule is error.

Comment (mudit2812) [reply pennylane/workflow/_capture_qnode.py:303]: Why does the lowering not fail for other primitives in that case?

Comment (albi3ro) [reply pennylane/workflow/_capture_qnode.py:303]: The other primitives aren't seen by `jax.jit`.  `jax.jit` only sees the execution of the jaxpr, not the capturing of it.

Comment (albi3ro) [reply pennylane/workflow/_capture_qnode.py:303]: We may also need to do this to the gradient primitives though.

Comment (mudit2812) [reply pennylane/workflow/_capture_qnode.py:303]: Thanks. Had a chat offline. I thought this would also impact other higher-order primitives, but `DefaultQubitInterpreter` unrolls those primitives, so we don't execute the jaxpr of other HOPs, which is why we don't need to do the same registration for them.

Approving.

**Review by mudit2812 — APPROVED** [review]

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -165,6 +165,7 @@ featuring a `simulate` function for simulating mixed states in analytic mode.
 * Execution with capture enabled now follows a new execution pipeline and natively passes the
   captured jaxpr to the device. Since it no longer falls back to the old pipeline, execution
   only works with a reduced feature set.
+  [(#6496)](https://github.com/PennyLaneAI/pennylane/pull/6596)
```
Comment (lillian542) [inline doc/releases/changelog-dev.md:168]:
```suggestion
  [(#6655)](https://github.com/PennyLaneAI/pennylane/pull/6655)
```

**Review by lillian542 — APPROVED** [review]
LGTM!
